2996_reflection/reflection.md

```diff
@@ -2495,29 +2630,55 @@ namespace std::meta {
 ```
 :::
 
-These metafunctions produces a reflection of the value returned by a call expression.
+These metafunctions produce a reflection of the result of a call expression.
+
+For the first overload: Letting `F` be the entity reflected by `target`, and `A@~0~@, A@~1~@, ..., A@~N~@` be the sequence of entities reflected by the values held by `args`: if the expression `F(A@~0~@, A@~1~@, ..., A@~N~@)` is a well-formed constant expression evaluating to a structural type that is not `void`, and if every value in `args` is a reflection of a value or object usable in constant expressions, then `reflect_invoke(target, args)` evaluates to a reflection of the result of `F(A@~0~@, A@~1~@, ..., A@~N~@)`. For all other invocations, `reflect_invoke(target, args)` is not a constant expression.
 
-For the first overload: Letting `F` be the entity reflected by `target`, and `A@~0~@, A@~1~@, ..., A@~N~@` be the sequence of entities reflected by the values held by `args`: if the expression `F(A@~0~@, A@~1~@, ..., A@~N~@)` is a well-formed constant expression evaluating to a type that is not `void`, and if every value in `args` is a reflection of a value or object usable in constant expressions, then `reflect_invoke(target, args)` evaluates to a reflection of the result of `F(A@~0~@, A@~1~@, ..., A@~N~@)`. For all other invocations, `reflect_invoke(target, args)` is not a constant expression.
+The second overload behaves the same as the first overload, except instead of evaluating `F(A@~0~@, A@~1~@, ..., A@~N~@)`, we require that `F` be a reflection of a template and evaluate `F<T@~0~@, T@~1~@, ..., T@~M~@>(A@~0~@, A@~1~@, ..., A@~N~@)`. This allows evaluating `reflect_invoke(^std::get, {std::meta::reflect_value(0)}, {e})` to evaluate to, approximately, `^std::get<0>([: e :])`.
 
-The second overload behaves the same as the first overload, except instead of evaluating `F(A@~0~@, A@~1~@, ..., A@~N~@)`, we require that `F` be a reflection of a template and evaluate `F<T@~0~@, T@~1~@, ..., T@~M~@>(A@~0~@, A@~1~@, ..., A@~N~@)`. This allows evaluating `reflect_invoke(^std::get, {reflect_result(0)}, {e})` to evaluate to, approximately, `^std::get<0>([: e :])`.
+If the returned reflection is of a value, the type of the reflected value shall not be an alias and shall be cv-unqualified unless it is of class type.
```

Comment:
This isn't a good use of "shall" - both because we're not in Core wording and the rule here is just what `reflect_invoke` _does_, rather than being a requirement on the program. It's just that when `reflect_invoke` gives you a reflection of the value, the type of that value isn't an alias type and, if the type is scalar, is cv-unqualified. 

Also what do you mean *if* it's a value? What else could it be?

Reply:
> This isn't a good use of "shall"

👍 , done. Used language similar to what's used for `reflect_value` below.

> Also what do you mean if it's a value? What else could it be?

Could be a reflection of an object (i.e., a consteval function that returns a reference).

```cpp
consteval int &second(std::pair<int, int> &p) {
    return p.second;
}


int main() {
    static std::pair<int, int> p;
    [:reflect_invoke(^second, {^p}):] = 42;

    return p.second;
}
```

(currently an implementation bug that causes this not to work, but it will work on godbolt tomorrow)